plugin/src/py/android_screenshot_tests/pull_screenshots.py

```diff
@@ -445,7 +450,8 @@ def main(argv):
                             opt_generate_png=opts.get('--generate-png'),
                             record=opts.get('--record'),
                             verify=opts.get('--verify'),
-                            adb_puller=SimplePuller(puller_args))
+                            adb_puller=SimplePuller(puller_args),
+                            device_name_calculator=DeviceNameCalculator())
```

Comment:
you mentioned that you're not using a DeviceNameCalculator by default, but this seems to suggest otherwise. Am I missing something?

Reply:
where I have mentioned that I'm not using a DeviceNameCalculator by default? I look at this PR and I can't see it.
I think that you are talking about this PR https://github.com/facebook/screenshot-tests-for-android/pull/113, which is closed because you requested some changes and this one was created with the changes instead of changing that.

Did you have a look at this one?
Do you have any news?

Reply:
okay yeah, we definitely need this to default to none, otherwise it's going to break existing users directory structures (and definitely some tools internally at FB :) ). The new format can be made opt-in, perhaps in the future we can turn it on by default if enough people think it's the right move 

(@xiphirx is doing most of the review, I was just scanning through this for obvious breakages)

Reply:
I was thinking about a NoOpDeviceNameCalculator which returns an empty string when name method is called. This way we don't work with None's. What do you think?

By default, NoOpDeviceNameCalculator is used but users can turn DeviceNameCalculator by setting a true multipleDevices variable in their build.gradle file.

What do you think about creating a new app example project with this multiple devices feature turned on?

Is it okey for you, @xiphirx @tdrhq?

Reply:
Yep, NoOpDeviceNameCalculator sounds like the right design (but: it might need some more changes because the device_name_calculator right now assumes that it's going into a subdirectory, but the default mode shouldn't go into a subdirectory)

new example app: that would be fantastic, go for it

Reply:
Hey! I've updated the PR with the changes requested. I took the freedom to write the example app in Kotlin.

About this 'device_name_calculator right now assumes that it's going into a subdirectory, but the default mode shouldn't go into a subdirectory', I think that you're talking about the change in recorder.py from os.mkdir to os.makedirs. Makedirs works also if you want to create a just one directory nor only with multiple directories.

@xiphirx @tdrhq If you have any news, please let me know!